File: main/client/src/mill/main/client/lock/FileTryLocked.java
package mill.main.client.lock;

class FileTryLocked extends FileLocked implements TryLocked {
  public FileTryLocked(java.nio.channels.FileLock lock) {
    super(lock);
  }

  public boolean isLocked() {
    return lock != null;
  }

  public void release() throws Exception {
    if (isLocked()) super.release();
  }
}

File: main/client/src/lock/FileTryLocked.scala
package mill.main.client.lock

class FileTryLocked(lock: java.nio.channels.FileLock) extends FileLocked(lock) with TryLocked {

  def isLocked: Boolean = lock != null

  @throws[Exception]
  override def release(): Unit = {
    if (isLocked) super.release()
  }
}
